woudc_data_registry/models.py

```diff
@@ -1163,12 +1250,113 @@ def sync(ctx):
+                            station_models, instrument_models, country_models,
+                            deployment_models, contributor_models):
+
+        """turn lists holding essential information into dictionaries with key
+        that is common with another list in order to allow for fast lookup"""
+        country_dict = {country_models[i].country_id: country_models[i]
+                        for i in range(0, len(country_models))}
+        instrument_dict = {instrument_models[i].name: instrument_models[i]
+                           for i in range(0, len(instrument_models))}
+        contributor_dict = {contributor_models[i].contributor_id:
+                            contributor_models[i]
+                            for i in range(0, len(contributor_models))}
+
+        """deployment_dict will have stations as
+        keys in order to help connect them to projects """
+        deployment_dict = {deployment_models[i].station_id:
+                           deployment_models[i]
+                           for i in range(0, len(deployment_models))}
+
+        """contribution dict used to check for duplicate contribution id"""
+        contribution_dict = {contribution_models[i].contribution_id:
+                             contribution_models[i]
+                             for i in range(0, len(contribution_models))}
+
+        for key in instrument_dict:
+
+            for station in station_models:
+
+                """station info"""
+                station_id = station.station_id
+                station_name = station.station_name_id
+
+                """country info"""
+                country_id = station.country_id
+                country_name_en = country_dict[country_id].name_en
+                country_name_fr = country_dict[country_id].name_fr
+
+                """instrument info"""
+                instrument_id = instrument_dict[key].instrument_id
+                instrument_name = instrument_dict[key].name
+                instrument_model = instrument_dict[key].model
+
+                start_date = instrument_dict[key].start_date
+                end_date = instrument_dict[key].end_date
+                dataset_id = instrument_dict[key].dataset_id
+
+                x = instrument_dict[key].x
+                y = instrument_dict[key].y
+                z = instrument_dict[key].z
+
+                """ get the contributor id first from
+                the deployment at the specific station"""
+                contributor_id = \
+                    deployment_dict[station.station_id].contributor_id
+
+                """ now access the project from contributor"""
+                project_id = contributor_dict[contributor_id].project_id
+
+                """form the contribution id by combining the
+                strings present in Contributions dependencies"""
+                contribution_id = project_id + ":" + dataset_id + ":" \
+                    + station_id + ":" + instrument_name
```

Comment:
Good choice of components to make up the ID! You can also use `':'.join([... list of components ... ])` for the same effect.